### PR TITLE
feat: add dnsPolicy support to Deployment

### LINKED ISSUE #365: Add dnsPolicy support
**Is your feature request related to a problem? Please describe.**
I'm using this chart for an application using hostNetworking (which is supported), so I would also like to configure dnsPolicy on the pod spec with ClusterFirstWithHostNet.

**Describe the solution you'd like**
Add support for dnsPolicy to Deployment (and perhaps job and cronjob, but those do not currently support dnsConfig either): a string property for the PodSpec.

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by tomjo**
Fixes #365 

If wanted, I can also add it to job and cronjob but perhaps then support for dnsConfig should be added there as well?

### CODE REVIEW

**Comment by aslafy-z** [comment]
I think adding `dnsConfig` too would be a nice addition. Also, I would prefer them to be added to deployments, jobs and cronjob at the same time. Can you do these changes?
Please use `make build-docs` to update the readme file.

File: application/values.yaml [hunk 1]
```diff
@@ -251,6 +251,9 @@ deployment:
     # options:
     # - name: ndots
     #   value: '1'
+  # -- (string) DNS Policy.
+  # @section -- Deployment Parameters
+  dnsPolicy: ClusterFirst
```
Comment (aslafy-z) [inline application/values.yaml:256]: I would use the kubernetes default by default, by setting this value to an empty string (`""`).

File: application/values.yaml [hunk 2]
```diff
@@ -251,6 +251,9 @@ deployment:
...
   # -- (object) Startup probe.
   # @default -- See below
```
Comment (aslafy-z) [inline application/values.yaml:258]: Please add dnsPolicy and dnsConfig to jobs and cronjobs sections.

**Review by aslafy-z — CHANGES_REQUESTED** [review]

**Review by aslafy-z — APPROVED** [review]

**Review by rasheedamir — APPROVED** [review]

**Comment by rasheedamir** [comment]
@aslafy-z you can merge it now

**Review by aslafy-z — APPROVED** [review]

**Pull request merged**